auth-external-api/src/main/java/uk/gov/di/authentication/external/lambda/UserInfoHandler.java

```diff
@@ -186,6 +186,8 @@ public APIGatewayProxyResponseEvent userInfoRequestHandler(APIGatewayProxyReques
                     accessToken.getValue());
         }
 
+        sessionService.save(userSession.setNewAccount(Session.AccountState.EXISTING));
```

Comment:
Is the logic here that as long as the request is successful, any account (including new accounts) are marked as existing at this point?

Could you do some combo (maybe both :-)) of:
* writing a bit more about why this behaviour is useful in the commit message
* call it out as a specific scenario in a test (updates a new account to existing when user info handler request processed correctly?)

Reply:
I have added a separate test case for this and tried to elaborate a bit more in the commit message. Let me know if commit message is too context-y and not explain-y enough 😅